integration/delete_command_compat_test.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+
+	"github.com/FerretDB/FerretDB/integration/setup"
+	"github.com/FerretDB/FerretDB/integration/shareddata"
+)
+
+// deleteCommandCompatTestCase describes delete compatibility test case.
+type deleteCommandCompatTestCase struct {
+	skip    string // optional, skip test with a specified reason
+	deletes bson.A // required
+}
+
+func testDeleteCommandCompat(t *testing.T, testCases map[string]deleteCommandCompatTestCase) {
+	t.Helper()
+
+	for name, tc := range testCases {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Helper()
+
+			if tc.skip != "" {
+				t.Skip(tc.skip)
+			}
+
+			t.Parallel()
+
+			// Use per-test setup because deletes modify data set.
+			ctx, targetCollections, compatCollections := setup.SetupCompat(t)
+
+			for i := range targetCollections {
+				targetCollection := targetCollections[i]
+				compatCollection := compatCollections[i]
+				t.Run(targetCollection.Name(), func(t *testing.T) {
+					t.Helper()
+
+					var targetRes, compatRes bson.D
+					targetErr := targetCollection.Database().RunCommand(ctx,
+						bson.D{
+							{"delete", targetCollection.Name()},
+							{"deletes", tc.deletes},
+						}).Decode(&targetRes)
+					compatErr := compatCollection.Database().RunCommand(ctx,
+						bson.D{
+							{"delete", compatCollection.Name()},
+							{"deletes", tc.deletes},
+						}).Decode(&compatRes)
+
+					if targetErr != nil {
+						t.Logf("Target error: %v", targetErr)
+						targetErr = UnsetRaw(t, targetErr)
+						compatErr = UnsetRaw(t, compatErr)
+						assert.Equal(t, compatErr, targetErr)
+					} else {
```

Comment:
Please check other places with `t.Logf("Target error: %v", targetErr)`. Notice the `return` instead of `} else {`, `AssertMatchesCommandError` instead of `assert.Equal`, etc

Reply:
Thanks for noticing. Updated that part. For the `else` part, it was copied from the other delete test and there it should be as is because we have to check the response in case some documents were deleted. I left a comment about it.